model/proto/api_v2.proto

```diff
@@ -64,6 +64,22 @@ service CollectorService {
     }
 }
 
+message GetTraceRequest {
+    string Id = 1;
+}
+
+message GetTraceResponse {
+    Trace trace = 1;
+}
```

Comment:
check the discussion in https://github.com/jaegertracing/jaeger/pull/1323. We have an internal grpc implementation of GetTrace, and our users have run into the issue of large traces exceeding grpc default message size. So I think we need to go with streaming APIs for methods that can return large data sets.

I am not sure what it would do to grpc_gateway, how does it map streaming calls to HTTP?

Also, it's been a while since the last time I checked, but if gRPC for JavaScript in the browser has advanced enough, then we won't even need grpc_gateway.

Reply:
> I am not sure what it would do to grpc_gateway, how does it map streaming calls to HTTP?

Seems like they have an open issue for sending chunked JSON responses - grpc-ecosystem/grpc-gateway/issues/562

> Also, it's been a while since the last time I checked, but if gRPC for JavaScript in the browser has advanced enough, then we won't even need grpc_gateway.

Apart from streaming-support (which is on their [roadmap](https://github.com/grpc/grpc-web/blob/master/ROADMAP.md#streaming-friendly-transport-implementation)) what other features would we need in the grpc-web library?

Reply:
Update: Went through the grpc-web readme again, and it mentions that server-side streaming is already supported. This makes it a candidate to be considered for the query UI.

Reply:
Great! We can always put a blocking/sync facade in front of grpc service if streaming to web is not supported, but since it's there already, we should go with streaming for the main API.